### ISSUE TITLE
Emoji shown double in input field in special case

### ISSUE THREAD

**Issue by individual-it** [signalapp/Signal-Android#7494]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
When an emoji is followed by a special character (tested with `?!,:`) and emoji+char are about to be shifted into a new line of the input field, the emoji is displayed twice.
Confused? Yes, a funny bug ;-)

### Steps to reproduce
- type an emoji directly followed by a `?`
- fill the space before the emoji with spaces or any other char till the `?` has to be shifted to the new line

**Actual result:** the emoji is displayed twice
**Expected result:** the emoji should be displayed only once

### Screenshots
still OK:
![screenshot_2018-03-10-21-31-38](https://user-images.githubusercontent.com/2425577/37244037-0fb0f3ba-24ab-11e8-8743-54161fc5f8dd.png)

emoji shown double:

![screenshot_2018-03-10-21-31-27](https://user-images.githubusercontent.com/2425577/37244038-12cb85ec-24ab-11e8-8128-d3fbb438a884.png)

### Device info

**Device:** GIONEE P2M
**Android version:** 5.1
**Signal version:** 4.16.9

**Comment by FeuRenard** [comment]
This is a duplicate of #5385.
